tests/grpc/utils/TestServer.ts

```diff
@@ -0,0 +1,18 @@
+import * as grpc from '@grpc/grpc-js';
+import * as utils from './index';
+
+// This is spawned as a background process for use in some NodeConnection.test.ts tests
+async function main() {
+  const authenticate = async (metaClient, metaServer = new grpc.Metadata()) =>
+    metaServer;
+  const [server, port] = await utils.openTestServer(authenticate);
+  process.stdout.write(`${port}`);
+  process.stdin.on('data', () => {
+    server.forceShutdown();
+  });
+}
+
+main().then(
+  () => {},
+  () => {},
+);
```

Comment:
What is this file? If it is a script, it should have a lower case name. Upper case names are always referred to things exporting classes.

But also why does this require a separate process during testing?

Reply:
The running the tests that use `testUnaryFail` and `testStreamFail` we need the service to kill it's own process. Thus we have a `testServer` in a separate process. I'll fix the name.

Reply:
Use this at the very end:

```ts
if (require.main === module) {
  (async () => {
    await main();
  })();
}

export default main;
```

Reply:
Added.